### ISSUE TITLE
parse_obj return type is SQLModel

### ISSUE THREAD

**Issue by phi-friday** [tiangolo/sqlmodel#317]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel

class test(SQLModel):
    attr: int
    desc: str

class real(test, table=True):
    val: float

test_inst: test = test(attr=1, desc='test_inst')
real_inst: real = real.parse_obj(test_inst.dict())
"""
(method) parse_obj: (obj: Any, update: Dict[str, Any] | None = None) -> SQLModel
Expression of type "SQLModel" cannot be assigned to declared type "real"
  "SQLModel" is incompatible with "real"PylancereportGeneralTypeIssues
"""
```

### Description

real.parse_obj return type should be real.

### Wanted Solution

```python
# sqlmodel/main.py

from typing import TypeVar
_T = TypeVar("_T", bound="SQLModel")

(...)

# line no.569
    @classmethod
    def parse_obj(
        cls: Type[_T], obj: Any, update: Optional[Dict[str, Any]] = None
    ) -> _T:
        obj = cls._enforce_dict_if_root(obj)
        # SQLModel, support update dict
        if update is not None:
            obj = {**obj, **update}
        # End SQLModel support dict
        return super().parse_obj(obj)
```

### Wanted Code

...

### Alternatives

It's being used temporarily like this.

```python
from typing import Any, TypeVar, cast

from sqlmodel import SQLModel

_T = TypeVar("_T", bound=SQLModel)


class fix_parse_obj_model(SQLModel):
    @classmethod
    def parse_obj(cls: type[_T], obj: Any, update: dict[str, Any] | None = None) -> _T:
        return cast(_T, super().parse_obj(obj, update))
```

### Operating System

Linux

### Operating System Details

OS: Arch Linux x86_64 
Host: 82FX Yoga Slim 7 Pro 14ITL5 
Kernel: 5.17.4-arch1-1 
Uptime: 3 hours, 32 mins 
Packages: 969 (pacman), 23 (flatpak) 
Shell: zsh 5.8.1 
Resolution: 2240x1400, 2560x1080 
DE: GNOME 42.0 
WM: Mutter 
WM Theme: Adwaita 
Theme: Arc [GTK2/3] 
Icons: Papirus [GTK2/3] 
Terminal: vscode 
CPU: 11th Gen Intel i5-1135G7 (8) @ 4.200GHz 
GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics] 
GPU: NVIDIA GeForce MX450 
Memory: 5423MiB / 15785MiB

### SQLModel Version

0.0.6

### Python Version

Python 3.10.4

### Additional Context

I'd like to do a "pull requests"
shame I'm not used to git yet...

**Comment by antont** [comment]
you can make a pull request by forking the repo to your own account and committing your improvement to a branch there. then there's a create pull request button.

the creator & maintainer has been busy apparently, but at least we've been pulling useful PRs to our fork by ourselves, not many though as the base version has worked mostly fine.

**Comment by tiangolo** [comment]
Thanks for the report and the fix @phi-friday ! :cake: 

This was solved in https://github.com/tiangolo/sqlmodel/pull/321, the fix will be available in SQLModel `0.0.7`, released in the next hours. :tada:
